### PR TITLE
Check JM style metadata as a relevant subset of overall metadata [BA-6170]

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: centaur/src/main/scala/centaur/api/CentaurCromwellClient.scala [hunk 1]
```diff
@@ -107,6 +107,21 @@ object CentaurCromwellClient extends StrictLogging {
...
+      "%executionEvents%",
+      "status",
+      "workflowName",
+      "submission",
+      "start",
+      "end",
+      "failures"
+    ]
+   */
+
```
Comment (cjllanwarne) [inline centaur/src/main/scala/centaur/api/CentaurCromwellClient.scala]: Remove

File: centaur/src/main/scala/centaur/test/formulas/TestFormulas.scala [hunk 2]
```diff
@@ -42,6 +42,7 @@ object TestFormulas extends StrictLogging {
     _ <- checkDescription(workflowDefinition, validityExpectation = Option(true))
     submittedWorkflow <- runSuccessfulWorkflow(workflowDefinition)
     metadata <- validateMetadata(submittedWorkflow, workflowDefinition)
+    _ <- validateJobManagerStyleMetadata(submittedWorkflow, metadata.value)
```
Comment (cjllanwarne) [inline centaur/src/main/scala/centaur/test/formulas/TestFormulas.scala]: Any other places this should go?

Comment (mcovarr) [reply centaur/src/main/scala/centaur/test/formulas/TestFormulas.scala]: I *think* it could go anywhere `runSuccessfulWorkflow` is being called?

Comment (cjllanwarne) [reply centaur/src/main/scala/centaur/test/formulas/TestFormulas.scala]: In fact, every time we record the metadata, we can check that it gets JM-ified properly.

File: centaur/src/main/scala/centaur/test/Test.scala [hunk 3]
```diff
@@ -647,6 +646,124 @@ object Operations extends StrictLogging {
...
+    "excludeKey" -> List("callCaching:hitFailures"),
+    "expandSubWorkflows" -> List("false")
+  )
+
+  // I chose to re-implement "process metadata for job manager's includeKeys/excludeKeys" and not re-use production.
+  // Reasoning is:
+  //  1. the ultra-specific JM case is a lot simpler to reason about and can be done in a few lines
+  //  2. it's nice to be able to assert general cases rather than singular specific examples
+  //  3. it's hopefully unlikely that the same bug will show up in two separate implementations
+  def setUpJmStyleMetadataExpectation(originalWorkflowMetadataJson: JsObject): JsObject = {
```
Comment (grsterin) [inline centaur/src/main/scala/centaur/test/Test.scala]: TOL: Method name seems a bit confusing to me. Wouldn't it be better to call it something like `convertOriginalMetadataIntoJmStyleMetadata` or `composeJmStyleMetadataFromOriginalMetadata`?

File: centaur/src/test/scala/centaur/test/metadata/JobManagerExpectationSetupSpec.scala [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package centaur.test.metadata
+
+import centaur.test.Operations
+import org.scalatest.{FlatSpec, Matchers}
+import spray.json._
+
+class JobManagerExpectationSetupSpec extends FlatSpec with Matchers {
```
Comment (grsterin) [inline centaur/src/test/scala/centaur/test/metadata/JobManagerExpectationSetupSpec.scala]: TOL: same thing: https://github.com/broadinstitute/cromwell/pull/5403/files#r376821353

File: centaur/src/main/scala/centaur/test/Test.scala (hunk 3 above)
Comment (cjllanwarne) [reply centaur/src/main/scala/centaur/test/Test.scala]: Sure, that works. I'll make the change

**Pull request merged**